src/js/view/controls/settings-menu.js

```diff
@@ -179,7 +180,15 @@ export function setupSubmenuListeners(settingsMenu, controlbar, model, api) {
             const mediaModel = model.get('mediaModel');
             onAudiotracksChanged(mediaModel, mediaModel, mediaModel.get('audioTracks'));
             onQualitiesChanged(model, mediaModel.get('levels'));
-            onPlaybackRatesChange(model, model.get('playbackRates'));
+            setupPlaybackRatesMenu(model, model.get('playbackRates'));
         }
     });
+
+    model.on('change:streamType', (changedModel, newStreamType, previousStreamType) => {
+        if (newStreamType === previousStreamType) {
```

Comment:
You shouldn't have to check for equality here, unless an artificial "change" event is being dispatched. `model.on('change:X` only fires when X is changed.